Documentation.docc/V4_API_Migration_guide.md

```diff
@@ -0,0 +1,193 @@
+| Purchases.ErrorCode.Code | See error handling below |
+| Purchases.Package | ``Package`` |
+| Purchases.PurchaserInfo | <strong>``CustomerInfo``</strong> |
+| Purchases.EntitlementInfos | ``EntitlementInfos`` |
+| Purchases.Transaction | ``StoreTransaction`` |
+| Purchases.EntitlementInfo | ``EntitlementInfo`` |
+| Purchases.PeriodType | ``PeriodType`` |
+| Purchases.Store | ``Store`` |
+| RCPurchaseOwnershipType | ``PurchaseOwnershipType`` |
+| RCAttributionNetwork | ``AttributionNetwork`` |
+| RCIntroEligibility | ``IntroEligibility`` |
+| RCIntroEligibilityStatus | ``IntroEligibilityStatus`` |
+| Purchases.LogLevel | ``LogLevel`` |
+| Purchases.Offerings | ``Offerings`` |
+| Purchases.PackageType | ``PackageType`` |
+| Package.product | ``Package/storeProduct`` |
+| Package.product.price: NSDecimalNumber | ``StoreProduct/price``: Decimal |
+| Package.localizedIntroductoryPriceString: String | ``Package/localizedIntroductoryPriceString``: String? |
+| RCDeferredPromotionalPurchaseBlock | ``DeferredPromotionalPurchaseBlock`` |
+| Purchases.PurchaseCompletedBlock | ``PurchaseCompletedBlock`` |
+| Purchases.ReceivePurchaserInfoBlock | <i>REMOVED</i> |
+| Purchases.ReceiveOfferingsBlock | <i>REMOVED</i> |
+| Purchases.ReceiveProductsBlock | <i>REMOVED</i> |
+| Purchases.PaymentDiscountBlock | <i>REMOVED</i> |
+| Purchases.RevenueCatBackendErrorCode | <i>REMOVED</i> |
+| Purchases.ErrorCode.operationAlreadyInProgressError | ``RevenueCat/ErrorCode/operationAlreadyInProgressForProductError`` |
+| Purchases.ErrorUtils | <i>REMOVED</i> |
+| ReadableErrorCodeKey | <i>REMOVED</i> |
+| RCFinishableKey | <i>REMOVED</i> |
+
+### Swift API changes
+
+| v3 | v4 |
+| ------------ | ------------------------------------- | 
+| invalidatePurchaserInfoCache | ``Purchases/invalidateCustomerInfoCache()`` |
+| logIn(_ appUserId:, _ completion:) | ``Purchases/logIn(_:completion:)`` |
+| createAlias(_ alias:, _ completion:) | ``Purchases/logIn(_:completion:)`` |
+| identify(_ appUserID:, _ completion:) | ``Purchases/logIn(_:completion:)`` |
+| reset(completion:) | ``Purchases/logOut(completion:)`` |
+| purchaserInfo(_ completion:) | ``Purchases/getCustomerInfo(completion:)`` |
+| offerings(_ completion:) | ``Purchases/getOfferings(completion:)`` |
+| products(_ productIdentifiers:, _ completion:) | ``Purchases/getProducts(_:completion:)`` |
+| purchaseProduct(_ product:, _ completion:) | ``Purchases/purchase(product:completion:)`` |
+| purchasePackage(_ package:, _ completion:) | ``Purchases/purchase(package:completion:)`` |
+| restoreTransactions(_ completion:) | ``Purchases/restorePurchases(completion:)`` |
+| syncPurchases(_ completion:) | ``Purchases/syncPurchases(completion:)`` |
+| paymentDiscount(for:product:completion:) | Can be accessed from ``StoreProduct/introductoryPrice`` |
+| purchaseProduct(_:discount:_) | ``Purchases/purchase(product:discount:completion:)`` |
+| purchasePackage(_:discount:_) | ``Purchases/purchase(package:discount:completion:)`` |
```

Comment:
for Swift types, I added links to the actual types so you can click on them if you're viewing this as Docc (through Xcode's package documentation). If we deployed this as Docc or a separate Jazzy article, the link should work as well. 
The main advantage, though, is that we get type checks from Xcode here so we don't mess up a symbol name